prog/base.rb

```diff
@@ -179,6 +179,19 @@ def hop(label)
     fail Hop.new(@strand.prog, @strand.label, {label: label, retval: nil})
   end
 
+  def register_deadline(deadline_target, deadline_in)
+    strand.stack.first["deadline_target"] = deadline_target
+    strand.stack.first["deadline_at"] = Time.now + deadline_in
+
+    # Sequel is only able to detect modifications if they are made
+    # using column's setter direcly like "strand.stack = ...". It
+    # does not detect above updates because we are updating elements
+    # of the stack, not the stack itself. Thus, we are letting it
+    # know that strand column is modified and the change needs to be
+    # persisted in the DB.
+    strand.modified!(:stack)
```

Comment:
We expect `strand.save_changes` or `strand.update` are called in some point after registering deadline, right?

Reply:
Yes, we do. It is called at Strand.unsynchronized_run for all FlowControl rescues except nap, which I think is a bug independent from this PR. For now, I added save_changes to rescue Prog::Base::Nap, though I'm inclined to refactor transaction block in Strand.unsynchronized_run (via future PR) because we do 3 different kind of database modifications in it and it is bit complex.

Reply:
Also moved the explanation to commit message.